backend/pkg/kafka/config_helper.go

```diff
@@ -146,12 +147,37 @@ func NewKgoConfig(cfg *config.Kafka, logger *zap.Logger, hooks kgo.Hook) ([]kgo.
 
 		// AWS MSK IAM
 		if cfg.SASL.Mechanism == config.SASLMechanismAWSManagedStreamingIAM {
-			mechanism := aws.Auth{
-				AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
-				SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
-				SessionToken: cfg.SASL.AWSMskIam.SessionToken,
-				UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
-			}.AsManagedStreamingIAMMechanism()
+			var mechanism sasl.Mechanism
+			if cfg.SASL.AWSMskIam.AccessKey != "" &&
+				cfg.SASL.AWSMskIam.SecretKey != "" {
+				// SessionToken is optional
+				mechanism = aws.Auth{
+					AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
+					SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
+					SessionToken: cfg.SASL.AWSMskIam.SessionToken,
+					UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
+				}.AsManagedStreamingIAMMechanism()
+			} else {
+				cfgVal, err := awsConfig.LoadDefaultConfig(
+					context.TODO(),
+					awsConfig.WithRegion(cfg.SASL.AWSMskIam.Region),
+				)
+				if err != nil {
+					return nil, err
+				}
+
+				creds, err := cfgVal.Credentials.Retrieve(context.TODO())
```

Comment:
Same here, let's use a ctx that times out in case we never get a response

Reply:
updated